### ISSUE TITLE
Difficulty getting flutter to work

### ISSUE THREAD

**Issue by abiodun005ng** [flutter/flutter#150497]
### Steps to reproduce

I did all my installation using Android Studio Jellifish, accepted all licenses, set proper home variables and all.

But when I run flutter doctor -v
- I have error with connected devices
- Visual Studio code is showing error that 'The flutter daemon has terminated';
- 
![Screenshot 2024-06-19 111814](https://github.com/flutter/flutter/assets/13048730/1921040f-212c-4c90-8049-79450c736868)

![Screenshot 2024-06-19 111830](https://github.com/flutter/flutter/assets/13048730/0b70b5b4-d608-4712-90a4-c8b9839ed0f7)
![Screenshot 2024-06-19 111932](https://github.com/flutter/flutter/assets/13048730/8f84fbe0-11b0-4a42-b5bc-febd57dfb972)
![Screenshot 2024-06-19 111950](https://github.com/flutter/flutter/assets/13048730/45aa4e77-9ff7-47db-9702-785a8fbe1caf)
![Screenshot 2024-06-19 111958](https://github.com/flutter/flutter/assets/13048730/7d2aad69-26d2-4a82-8f36-05a1bade9574)
![Uploading Screenshot (6).png…]()

### Actual results

[☠] Connected device (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know
      about this issue at https://github.com/flutter/flutter/issues.
    ✗ Error: Unable to run "adb", check your Android SDK installation and ANDROID_HOME environment variable:
      C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\adb.exe
    • #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
      #1      AndroidDevices.pollingGetDevices (package:flutter_tools/src/android/android_device_discovery.dart:75:7)
      <asynchronous suspension>
      #2      PollingDeviceDiscovery._populateDevices (package:flutter_tools/src/device.dart:547:36)
      <asynchronous suspension>
      #3      Future.wait.<anonymous closure> (dart:async/future.dart:524:21)
      <asynchronous suspension>
      #4      DeviceManager.refreshAllDevices (package:flutter_tools/src/device.dart:217:40)
      <asynchronous suspension>
      #5      DeviceValidator.validate (package:flutter_tools/src/doctor.dart:741:34)
      <asynchronous suspension>
      #6      Future.any.onValue (dart:async/future.dart:618:5)
      <asynchronous suspension>

### Logs

[11:17:31 AM] [General] [Info] Searching for SDKs...
[11:17:31 AM] [General] [Info] Environment PATH:
[11:17:31 AM] [General] [Info]     C:\WINDOWS\system32
[11:17:31 AM] [General] [Info]     C:\WINDOWS
[11:17:31 AM] [General] [Info]     C:\WINDOWS\System32\Wbem
[11:17:31 AM] [General] [Info]     C:\WINDOWS\System32\WindowsPowerShell\v1.0\
[11:17:31 AM] [General] [Info]     C:\WINDOWS\System32\OpenSSH\
[11:17:31 AM] [General] [Info]     C:\Program Files\Git\cmd
[11:17:31 AM] [General] [Info]     C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\
[11:17:31 AM] [General] [Info]     C:\flutter\bin
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Programs\Microsoft VS Code\bin
[11:17:31 AM] [General] [Info]     C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts
[11:17:31 AM] [General] [Info]     C:\Program Files\JetBrains\PyCharm 2024.1.2\bin
[11:17:31 AM] [General] [Info] Took 216ms to search for projects (5 levels)
[11:17:31 AM] [General] [Info] Took 4ms to filter out excluded projects (0 exclusion rules)
[11:17:31 AM] [General] [Info] Found Flutter project at C:\devs\flutter\learningdart:
			Mobile? true
			Web? false
			Create Trigger? false
			Flutter Repo? false
[11:17:31 AM] [General] [Info] Searching for flutter.bat
[11:17:31 AM] [General] [Info]     Looking for flutter.bat in:
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart -> C:\devs\flutter\learningdart\bin
[11:17:31 AM] [General] [Info]         C:\flutter\bin\
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart\.flutter
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart\.flutter -> C:\devs\flutter\learningdart\.flutter\bin
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart\vendor\flutter
[11:17:31 AM] [General] [Info]         C:\devs\flutter\learningdart\vendor\flutter -> C:\devs\flutter\learningdart\vendor\flutter\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\flutter-sdk
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\flutter-sdk -> C:\Users\Zbook\flutter-sdk\bin
[11:17:31 AM] [General] [Info]         /google/flutter
[11:17:31 AM] [General] [Info]         /google/flutter -> \google\flutter\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\system32
[11:17:31 AM] [General] [Info]         C:\WINDOWS\system32 -> C:\WINDOWS\system32\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS
[11:17:31 AM] [General] [Info]         C:\WINDOWS -> C:\WINDOWS\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\Wbem
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\Wbem -> C:\WINDOWS\System32\Wbem\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\WindowsPowerShell\v1.0\
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\WindowsPowerShell\v1.0\ -> C:\WINDOWS\System32\WindowsPowerShell\v1.0\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\OpenSSH\
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\OpenSSH\ -> C:\WINDOWS\System32\OpenSSH\bin
[11:17:31 AM] [General] [Info]         C:\Program Files\Git\cmd
[11:17:31 AM] [General] [Info]         C:\Program Files\Git\cmd -> C:\Program Files\Git\cmd\bin
[11:17:31 AM] [General] [Info]         C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\
[11:17:31 AM] [General] [Info]         C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\ -> C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\bin
[11:17:31 AM] [General] [Info]         C:\flutter\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools -> C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools -> C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\ -> C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps -> C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Microsoft VS Code\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts -> C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts\bin
[11:17:31 AM] [General] [Info]         C:\Program Files\JetBrains\PyCharm 2024.1.2\bin
[11:17:31 AM] [General] [Info]     Found at:
[11:17:31 AM] [General] [Info]         C:\flutter\bin\
[11:17:31 AM] [General] [Info]         C:\flutter\bin
[11:17:31 AM] [General] [Info]     Candidate paths to be post-filtered:
[11:17:31 AM] [General] [Info]         C:\flutter\bin\ -> C:\flutter
[11:17:31 AM] [General] [Info]         C:\flutter\bin -> C:\flutter
[11:17:31 AM] [General] [Info]     Found at C:\flutter\bin\ -> C:\flutter
[11:17:31 AM] [General] [Info]     Returning SDK path C:\flutter for flutter.bat
[11:17:31 AM] [General] [Info] Searching for dart.exe
[11:17:31 AM] [General] [Info]     Looking for dart.exe in:
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk -> C:\flutter\bin\cache\dart-sdk\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\system32
[11:17:31 AM] [General] [Info]         C:\WINDOWS\system32 -> C:\WINDOWS\system32\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS
[11:17:31 AM] [General] [Info]         C:\WINDOWS -> C:\WINDOWS\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\Wbem
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\Wbem -> C:\WINDOWS\System32\Wbem\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\WindowsPowerShell\v1.0\
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\WindowsPowerShell\v1.0\ -> C:\WINDOWS\System32\WindowsPowerShell\v1.0\bin
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\OpenSSH\
[11:17:31 AM] [General] [Info]         C:\WINDOWS\System32\OpenSSH\ -> C:\WINDOWS\System32\OpenSSH\bin
[11:17:31 AM] [General] [Info]         C:\Program Files\Git\cmd
[11:17:31 AM] [General] [Info]         C:\Program Files\Git\cmd -> C:\Program Files\Git\cmd\bin
[11:17:31 AM] [General] [Info]         C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\
[11:17:31 AM] [General] [Info]         C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\ -> C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\bin
[11:17:31 AM] [General] [Info]         C:\flutter\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools -> C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools -> C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\ -> C:\Users\Zbook\AppData\Local\Programs\Python\Launcher\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps -> C:\Users\Zbook\AppData\Local\Microsoft\WindowsApps\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Microsoft VS Code\bin
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts
[11:17:31 AM] [General] [Info]         C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts -> C:\Users\Zbook\AppData\Local\Programs\Python\Python312\Scripts\bin
[11:17:31 AM] [General] [Info]         C:\Program Files\JetBrains\PyCharm 2024.1.2\bin
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk -> C:\flutter\bin\cache\dart-sdk\bin
[11:17:31 AM] [General] [Info]     Found at:
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk -> C:\flutter\bin\cache\dart-sdk\bin
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk -> C:\flutter\bin\cache\dart-sdk\bin
[11:17:31 AM] [General] [Info]     Candidate paths to be post-filtered:
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk
[11:17:31 AM] [General] [Info]         C:\flutter\bin\cache\dart-sdk
[11:17:31 AM] [General] [Info]     Found at C:\flutter\bin\cache\dart-sdk
[11:17:31 AM] [General] [Info]     Returning SDK path C:\flutter\bin\cache\dart-sdk for dart.exe
[11:17:31 AM] [General] [Info] Experiment random number is 61 for experiment 'sdkDaps'. Experiment is enabled for <= 100
[11:17:31 AM] [General] [Info] Experiment 'sdkDaps' is activating (not first time)
[11:17:31 AM] [General] [Info] Experiment random number is 61 for experiment 'sdkDaps'. Experiment is enabled for <= 100
[11:17:31 AM] [General] [Info] Experiment 'sdkDaps' is activating (not first time)
[11:17:31 AM] [General] [Info] Experiment random number is 61 for experiment 'sdkDaps'. Experiment is enabled for <= 10
[11:17:31 AM] [General] [Info] Experiment 'sdkDaps' does not apply and will not be activated
[11:17:31 AM] [General] [Info] !! ⚠️ PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING ⚠️ !!
Dart Code extension: 3.90.0
Flutter extension: 3.90.0 (not activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.90.1

Workspace type: Flutter (LSP)

Dart (3.4.3): C:\flutter\bin\cache\dart-sdk
Flutter (3.22.2): C:\flutter (No device)
[11:17:31 AM] [DartToolingDaemon] [Info] Spawning C:\flutter\bin\cache\dart-sdk\bin\dart.exe with args ["tooling-daemon","--machine"]
[11:17:31 AM] [DartToolingDaemon] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code","FLUTTER_ROOT":"C:\\flutter"}}
[11:17:31 AM] [DartToolingDaemon] [Info]     PID: 13928
[11:17:31 AM] [FlutterDaemon] [Info] Spawning C:\flutter\bin\flutter.bat with args ["daemon"]
[11:17:31 AM] [FlutterDaemon] [Info] ..  in C:\flutter
[11:17:31 AM] [FlutterDaemon] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code","FLUTTER_ROOT":"C:\\flutter"}}
[11:17:31 AM] [FlutterDaemon] [Info]     PID: 13928
[11:17:31 AM] [General] [Info] Returning cached results for project search
[11:17:31 AM] [General] [Info] Checking 1 projects for supported platforms
[11:17:31 AM] [Analyzer] [Info] Spawning C:\flutter\bin\cache\dart-sdk\bin\dart.exe with args ["language-server","--protocol=lsp","--client-id=VS-Code","--client-version=3.90.0"]
[11:17:31 AM] [Analyzer] [Info]     PID: 10112
[11:17:31 AM] [General] [Info] Returning cached results for project search
[11:17:31 AM] [General] [Info] Activating Flutter extension for Flutter project...
[11:17:31 AM] [General] [Info] Version last used for Pub is 3.4.3 (3.4.0), current is 3.4.3 (3.4.0)
[11:17:31 AM] [General] [Info] Found 0 folders requiring "pub get" or "pub upgrade":
[11:17:31 AM] [Analyzer] [Info] ==> Content-Length: 6482
[11:17:31 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","id":0,"method":"initialize","params":{"processId":13928,"clientInfo":{"name":"Visual Studio Code","version":"1.90.1"},"locale":"en","rootPath":"c:\\devs\\flutter\\learningdart","rootUri":"file:///C%3A/devs/flutter/learningdart","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional","normalizesLineEndings":true,"changeAnnotationSupport":{"groupsOnLabel":…
[11:17:32 AM] [General] [Info] Loaded 20000 package names from vscode-userdata:/c%3A/Users/Zbook/AppData/Roaming/Code/User/globalStorage/dart-code.dart-code/package_cache.json
[11:17:32 AM] [General] [Info] !! ⚠️ PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING ⚠️ !!
Dart Code extension: 3.90.0
Flutter extension: 3.90.0 (activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.90.1

Workspace type: Flutter (LSP)

Dart (3.4.3): C:\flutter\bin\cache\dart-sdk
Flutter (3.22.2): C:\flutter (No device)
[11:17:32 AM] [DartToolingDaemon] [Info] <== {"tooling_daemon_details":{"uri":"ws://127.0.0.1:63221/QNlNDCF8eWlUFinU","trusted_client_secret":"BENKBRgZrsl0s1ij"}}
[11:17:32 AM] [DevTools] [Info] Spawning C:\flutter\bin\cache\dart-sdk\bin\dart.exe with args ["devtools","--machine","--allow-embedding","--dtd-uri","ws://127.0.0.1:63221/QNlNDCF8eWlUFinU"]
[11:17:32 AM] [DevTools] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code","FLUTTER_ROOT":"C:\\flutter"}}
[11:17:32 AM] [DevTools] [Info]     PID: 13928
[11:17:32 AM] [DartToolingDaemon] [Info] Connecting to DTD at ws://127.0.0.1:63221/QNlNDCF8eWlUFinU...
[11:17:33 AM] [DartToolingDaemon] [Info] Connected to DTD
[11:17:33 AM] [DartToolingDaemon] [Info] ==> {"id":"1","jsonrpc":"2.0","method":"FileSystem.setIDEWorkspaceRoots","params":{"secret":"BENKBRgZrsl0s1ij","roots":["file:///c%3A/devs/flutter/learningdart"]}}
[11:17:33 AM] [Analyzer] [Info] <== Content-Length: 807
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:17:33 AM] [Analyzer] [Info] <== {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"executeCommandProvider":{"commands":["dart.edit.sortMembers","dart.edit.organizeImports","dart.edit.fixAll","dart.edit.fixAllInWorkspace.preview","dart.edit.fixAllInWorkspace","dart.edit.sendWorkspaceEdit","refactor.perform","refactor.validate","dart.logAction","dart.refactor.convert_all_formal_parameters_to_named","dart.refactor.convert_selected_formal_parameters_to_named","dart.refactor.move_selected_formal_parameters_left","dart.refactor.…
[11:17:33 AM] [Analyzer] [Info] ==> Content-Length: 86
[11:17:33 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","method":"initialized","params":{},"clientRequestTime":1718792253302}
[11:17:33 AM] [DartToolingDaemon] [Info] <== {"jsonrpc":"2.0","result":{"type":"Success"},"id":"1"}
[11:17:33 AM] [Analyzer] [Info] <== Content-Length: 168
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:17:33 AM] [Analyzer] [Info] <== {"id":1,"jsonrpc":"2.0","method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///C:/devs/flutter/learningdart","section":"dart"},{"section":"dart"}]}}
[11:17:33 AM] [Analyzer] [Info] ==> Content-Length: 7565
[11:17:33 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","id":1,"result":[{"additionalAnalyzerFileExtensions":[],"analysisExcludedFolders":[],"analyzeAngularTemplates":true,"analyzerAdditionalArgs":[],"analyzerVmAdditionalArgs":[],"analyzerDiagnosticsPort":null,"analyzerPath":null,"analyzerSshHost":null,"analyzerVmServicePort":null,"includeDependenciesInWorkspaceSymbols":true,"notifyAnalyzerErrors":true,"onlyAnalyzeProjectsWithOpenFiles":false,"showTodos":true,"showExtensionRecommendations":true,"useLegacyAnalyzerProtocol":false,"…
[11:17:33 AM] [Analyzer] [Info] <== Content-Length: 5476
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:17:33 AM] [Analyzer] [Info] <== {"id":2,"jsonrpc":"2.0","method":"client/registerCapability","params":{"registrations":[{"id":"0","method":"textDocument/prepareCallHierarchy","registerOptions":{"documentSelector":[{"language":"dart","scheme":"file"}]}},{"id":"1","method":"textDocument/codeAction","registerOptions":{"codeActionKinds":["source","source.organizeImports","source.fixAll","source.sortMembers","quickfix","refactor"],"documentSelector":[{"language":"dart","scheme":"file"}]}},{"id":"2","method":"textDocument/codeLe…
[11:17:33 AM] [Analyzer] [Info] ==> Content-Length: 72
[11:17:33 AM] [Analyzer] [Info] ==> {"jsonrpc":"2.0","id":2,"result":null,"clientRequestTime":1718792253797}
[11:17:34 AM] [DevTools] [Info] <== {"event":"server.started","method":"server.started","params":{"host":"127.0.0.1","port":9100,"pid":16272,"protocolVersion":"1.2.0"}}
[11:17:34 AM] [Analyzer] [Info] <== Content-Length: 97
Content-Type: application/vscode-jsonrpc; charset=utf-8
[11:17:34 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.connected","params":{"version":"0.6.1","pid":17428}}]
[11:17:34 AM] [FlutterDaemon] [Info] ==> [{"id":"1","method":"device.enable"}]
[11:17:34 AM] [FlutterDaemon] [Info] ==> [{"id":"2","method":"emulator.getEmulators"}]
[11:17:34 AM] [FlutterDaemon] [Info] ==> [{"id":"3","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"C:\\devs\\flutter\\learningdart"}}]
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.logMessage","params":{"level":"status","message":"Starting device daemon..."}}]
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"id":"1"}]
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"id":"3","result":{"platforms":["web","android","ios","linux","macos","windows"],"platformTypes":{"web":{"isSupported":true},"android":{"isSupported":true},"ios":{"isSupported":true},"linux":{"isSupported":true},"macos":{"isSupported":true},"windows":{"isSupported":true},"fuchsia":{"isSupported":false,"reasons":[{"reasonText":"the Fuchsia feature is not enabled","fixText":"Run \"flutter config --enable-fuchsia\"","fixCode":"config"},{"reasonText":"the Fuchsia platform is not enabled for th…
[11:17:35 AM] [General] [Info] Supported platforms for the workspace are web, android, ios, linux, macos, windows
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"windows","name":"Windows","platform":"windows-x64","emulator":false,"category":"desktop","platformType":"windows","ephemeral":false,"emulatorId":null,"sdk":"Microsoft Windows [Version 10.0.22631.3737]","isConnected":true,"connectionInterface":"attached","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":true,"hardwareRendering":true,"startPaused":true}}}]
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"event":"daemon.logMessage","params":{"level":"error","message":"Unable to run \"adb\", check your Android SDK installation and ANDROID_HOME environment variable: C:\\Users\\Zbook\\AppData\\Local\\Android\\sdk\\platform-tools\\adb.exe"}}]
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"id":"2","result":[{"id":"INFO    | Storing crashdata in: C:\\Users\\Zbook\\AppData\\Local\\Temp\\\\AndroidEmulator\\emu-crash-34.2.15.db, detection is enabled for process: 9412","name":"INFO    | Storing crashdata in: C:\\Users\\Zbook\\AppData\\Local\\Temp\\\\AndroidEmulator\\emu-crash-34.2.15.db, detection is enabled for process: 9412","category":"mobile","platformType":"android"},{"id":"Pixel_3a_API_34","name":"Pixel 3a API 34","category":"mobile","platformType":"android"},{"id":"Pixel_…
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"chrome","name":"Chrome","platform":"web-javascript","emulator":false,"category":"web","platformType":"web","ephemeral":false,"emulatorId":null,"sdk":"Google Chrome 126.0.6478.61","isConnected":true,"connectionInterface":"attached","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":false,"hardwareRendering":true,"startPaused":true}}}]
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()
[11:17:35 AM] [FlutterDaemon] [Info] <== [{"event":"device.added","params":{"id":"edge","name":"Edge","platform":"web-javascript","emulator":false,"category":"web","platformType":"web","ephemeral":false,"emulatorId":null,"sdk":"Microsoft Edge 126.0.2592.56","isConnected":true,"connectionInterface":"attached","capabilities":{"hotReload":true,"hotRestart":true,"screenshot":false,"fastStart":false,"flutterExit":false,"hardwareRendering":true,"startPaused":true}}}]
[11:17:35 AM] [General] [Info] Returning cached promise for getSupportedPlatforms()

### Flutter Doctor output

[✓] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.22631.3737], locale en-NG)
    • Flutter version 3.22.2 on channel stable at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (2 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\Zbook\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = C:\Users\Zbook\AppData\Local\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.1)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.10.34928.147
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2023.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)

[✓] VS Code (version 1.90.1)
    • VS Code at C:\Users\Zbook\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.90.0

[☠] Connected device (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know
      about this issue at https://github.com/flutter/flutter/issues.
    ✗ Error: Unable to run "adb", check your Android SDK installation and ANDROID_HOME environment variable:
      C:\Users\Zbook\AppData\Local\Android\sdk\platform-tools\adb.exe
    • #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
      #1      AndroidDevices.pollingGetDevices (package:flutter_tools/src/android/android_device_discovery.dart:75:7)
      <asynchronous suspension>
      #2      PollingDeviceDiscovery._populateDevices (package:flutter_tools/src/device.dart:547:36)
      <asynchronous suspension>
      #3      Future.wait.<anonymous closure> (dart:async/future.dart:524:21)
      <asynchronous suspension>
      #4      DeviceManager.refreshAllDevices (package:flutter_tools/src/device.dart:217:40)
      <asynchronous suspension>
      #5      DeviceValidator.validate (package:flutter_tools/src/doctor.dart:741:34)
      <asynchronous suspension>
      #6      Future.any.onValue (dart:async/future.dart:618:5)
      <asynchronous suspension>

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

**Comment by huycozy** [comment]
Can you check [this](https://github.com/flutter/flutter/issues/72101#issuecomment-1185621903) to see if it helps?

**Comment by abiodun005ng** [comment]
It's not working, I have uninstalled, reinstalled the SDK platform tools, and restarted my system. I am still having the same issue.

**Comment by abiodun005ng** [comment]
![Screenshot 2024-06-19 135715](https://github.com/flutter/flutter/assets/13048730/bbd2c960-60c1-425a-ba26-4e059b716649)
![Screenshot 2024-06-19 135733](https://github.com/flutter/flutter/assets/13048730/54223af2-1b84-47a8-9ce3-8f7feabe05d1)
![Screenshot 2024-06-19 135800](https://github.com/flutter/flutter/assets/13048730/5b56a37d-ec3a-4a13-8929-0e5e570e2cbd)

**Comment by abiodun005ng** [comment]
![Screenshot 2024-06-19 135939](https://github.com/flutter/flutter/assets/13048730/02b418e0-d88a-49d5-89c8-12fe32fc894d)

**Comment by huycozy** [comment]
What about `adb devices` command? Have you tried it to see if `adb` command works well first?

You can check https://github.com/flutter/flutter/issues/142996#issuecomment-1931850239  and follow his steps to see if you also have the same issue.

**Comment by abiodun005ng** [comment]
I have been able to resolve this, practically, one of the fixes would have done that but still visual studio is repeating this error in it's terminal and also 'deamon terminated' error

![image](https://github.com/flutter/flutter/assets/13048730/fc337877-6fea-42d1-a165-6e1359e11859)
![image](https://github.com/flutter/flutter/assets/13048730/bc175d29-ffd4-4fa1-951b-b0d15b319798)

**Comment by huycozy** [comment]
Can you try again on a standalone PowerShell window (outside VSCode)? Or try restart VSCode to make sure the terminal session is reset after you fix for adb.

**Comment by abiodun005ng** [comment]
@huycozy , it's been good having you but all this solution has failed. However, I was able to find a way around it. I uninstall Android Studio, Visual Studio, visual studio code, and all their packages. I restarted my system, did a clean installation and everything works fine. The only comma is that I now run Visual Studio code with administrator privilege to work properly. 

Thanks so much for your responses so far. 

![image](https://github.com/flutter/flutter/assets/13048730/536cdff3-f0f7-4eb4-a94f-5ef2d2d2cc09)
![image](https://github.com/flutter/flutter/assets/13048730/a848edfc-7c7a-46b6-bdb3-3fdf897c6086)
